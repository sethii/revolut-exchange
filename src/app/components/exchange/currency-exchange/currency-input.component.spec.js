import React from 'react';
import renderer from 'react-test-renderer';
import { toClass } from 'recompose';
import { mount } from 'enzyme';

import { CurrencyInput } from './currency-input.component';

describe('Currency input component', () => {
  it('renders currency input', () => {
    const amount= '1234';
    const onChange = jest.fn();
    const prefix = "+";

    const CurrencyInputComponent = toClass(CurrencyInput);

    expect(renderer.create(<CurrencyInputComponent
      amount={amount}
      prefix={prefix}
      onChange={onChange}
    />).toJSON()).toMatchSnapshot();
  });

  it('it updates input when valid character pressed', () => {
    const amount= "1234";
    const onChange = jest.fn();
    const prefix = "+";

    const CurrencyInputComponent = toClass(CurrencyInput);

    const wrapper = mount(<CurrencyInputComponent
      amount={amount}
      prefix={prefix}
      onChange={onChange}
    />);

    wrapper.find('input').simulate('keyDown', { key: '1', keyCode: 20 });

    expect(onChange).toBeCalledWith('12341');
  });

  it('it does not updates input when invalid character pressed', () => {
    const amount= "1234";
    const onChange = jest.fn();
    const prefix = "+";

    const CurrencyInputComponent = toClass(CurrencyInput);

    const wrapper = mount(<CurrencyInputComponent
      amount={amount}
      prefix={prefix}
      onChange={onChange}
    />);

    wrapper.find('input').simulate('keyDown', { key: 'a', keyCode: 20 });

    expect(onChange).not.toBeCalled();
  });
});
