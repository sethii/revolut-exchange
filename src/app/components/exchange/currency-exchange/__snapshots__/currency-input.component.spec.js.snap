// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currency input component renders currency input 1`] = `
.glamor-1 {
  outline: none;
  border: none;
  padding: 10px 0px;
  font-size: 20px;
  text-align: right;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: transparent;
}

.glamor-1 ::-moz-selection {
  background: transparent;
}

.glamor-1 ::selection {
  background: transparent;
}

<input
  className="glamor-0 glamor-1"
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="0"
  type="text"
  value="+1234"
/>
`;
